src/org/freenetproject/freemail/wot/WoTConnectionImpl.java

```diff
@@ -272,28 +269,27 @@ private Message sendBlocking(final Message msg, Set<String> expectedMessageTypes
 		//Synchronize on pluginTalker so only one message can be sent at a time
 		final Message retValue;
 		Timer requestTimer;
-		synchronized(this) {
-			synchronized(replyLock) {
-				requestTimer = Timer.start();
 
-				assert (reply == null) : "Reply was " + reply;
-				reply = null;
+		synchronized(replyLock) {
```

Comment:
Holy crap, this is just horrible. Have you thought about ripping that code apart a bit and do it right? :/